db/migrate/20230611114027_add_index_to_mail_dispatch_rules_account_id.rb

```diff
@@ -3,6 +3,8 @@ class AddIndexToMailDispatchRulesAccountId < ActiveRecord::Migration[5.2]
 
   def change
     index_options = System::Database.postgres? ? { algorithm: :concurrently } : {}
-    add_index :mail_dispatch_rules, :account_id, index_options
+    add_index :mail_dispatch_rules, [:account_id, :system_operation_id], index_options
+
+    remove_index :mail_dispatch_rules, column: [:system_operation_id, :account_id]
```

Comment:
I'm not too much concerned about reverting. But I'm not sure this will work for reverting given that `index_options` would not be passed for rollback.

Other than that looks great!

Reply:
Well, I guess we can split in `up`/`down` to keep the index option for postgres